src/utility/Auth.js

```diff
@@ -10,28 +12,55 @@ export function useAuth() {
 
 export function AuthProvider({ children }) {
   const [user, setUser] = useState(null)
+  const [loading, setLoading] = useState(true)
 
   useEffect(() => {
     return firebase.auth().onAuthStateChanged(async currUser => {
+      if (!currUser) {
+        setLoading(false)
+        return
+      }
+      const status = await getUserStatus(currUser)
+      currUser.status = status
       setUser(currUser)
+      setLoading(false)
     })
   })
 
-  return (
+  return loading ? (
+    <>Loading...</>
+  ) : (
     <AuthContext.Provider value={{ isAuthed: !!user, user, setUser }}>
       {children}
     </AuthContext.Provider>
   )
 }
 
+const handleUser = async (setUser, setLocation) => {
+  const user = firebase.auth().currentUser
+  const status = await getUserStatus(user)
+  user.status = status
+  setUser(user)
+  setLocation(getRedirectUrl(status))
+}
+
+export const getRedirectUrl = status => {
+  switch (status) {
+    case applicantStatus.attending:
+      return '/judging'
+    case applicantStatus.applied || applicantStatus.accepted:
+    case applicantStatus.inProgress || applicantStatus.new:
```

Comment:
nitpick: ik you added these cases because these statuses will likely be used for other routes at some point but i don't think they're needed (and we can add them when it comes to it?)

Reply:
@panjenny0 do we have any plans for different routes for hackers that have already applied or have already been accepted? if not, I'll remove these extra variables.

Reply:
For hackers who have applied (applied, accepted, rejected, waitlisted), the plan is to redirect them to `/application` -  I think it's ok if you remove this case `case applicantStatus.applied || applicantStatus.accepted:`.
For people who are new or inProgress, it should take them to their first incomplete question (or a new form if they're new) - it would be great if you can add this as well, feel free to do it in a future PR!

Reply:
Good idea, I think I'll add these changes in a future PR like you suggested!